sql/core/src/test/scala/org/apache/spark/sql/execution/FileSourceScanExecSuite.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution
+
+import org.apache.spark.sql.test.SharedSQLContext
+
+class FileSourceScanExecSuite extends SharedSQLContext {
+  test("FileSourceScanExec should be canonicalizable on executor side") {
```

Comment:
I'd like to put this test in `QueryPlanSuite`, with name `SPARK-XXXX: query plans can be serialized and deserialized`.

In the test we don't need to trigger a job, just call `spark.env.serializer` to serialize and deserialize the `FileSourceScanExec`

Reply:
There's few things bothering for that actually - it's kind of messy to create `FileSourceScanExec` without `SparkSession` (and also without other utils from `SharedSQLContext`), and `QueryPlanSuite` is under `catalyst` whereas this plan itself is under `execution` in SQL core.

And, I actually believe this PR more targets to make the plan canonicalizable after it's de/serialized since this plan itself is serializable and deserializable already but it's not canonicalizable after that.

Let me try to clean up based on your comment anyway.

Reply:
I think I can actually put this under `SparkPlanSuite`. Let me put this it in.

Reply:
I found `SparkPlanSuite` could be another place to add to address your comment. Let me stick to `FileSourceScanExec` but please let me know if you prefer this please. I don't mind changing it.

Reply:
`SparkPlanSuite` SGTM